### PR TITLE
EZP-24408: Use REST API discovery service

### PR DESCRIPTION
**Pull request by yannickroger**
Link: https://jira.ez.no/browse/EZP-24408

## Description
Needs https://github.com/ezsystems/ez-js-rest-client/pull/60 to work as it implements the new feature introduced by it in the PlatformUIBundle. It is using the newly exposed discovery service.

## Tests
Manual and updated unit tests

### CODE REVIEW

**Comment by yannickroger** [comment]
Ready for review.
ping @mhyndle @dpobel @StephaneDiot

**Comment by StephaneDiot** [comment]
+1

File: Tests/js/models/assets/ez-locationmodel-tests.js [hunk 1]
```diff
@@ -71,27 +71,25 @@ YUI.add('ez-locationmodel-tests', function (Y) {
...
-                run: function (cb) {
-                    cb(false, loadRootResponse);
+            Mock.expect(capiMock, {
+                method: 'getDiscoveryService',
+                returns: discoveryServiceMock,
+            });
+
+            Mock.expect(discoveryServiceMock, {
+                method: 'getInfoObject',
+                args: [Mock.Value.String, Mock.Value.Function],
```
Comment (dpobel) [inline Tests/js/models/assets/ez-locationmodel-tests.js]: `Mock.Value.String` is not enough precise, I guess this should be `"Trash"` here

File: Tests/js/models/assets/ez-locationmodel-tests.js [hunk 2]
```diff
@@ -119,19 +117,19 @@ YUI.add('ez-locationmodel-tests', function (Y) {
...
+                returns: discoveryServiceMock,
             });
 
-            Mock.expect(contentServiceMock, {
-                method: 'loadRoot',
-                args: [Mock.Value.Function],
-                run: function (cb) {
+            Mock.expect(discoveryServiceMock, {
+                method: 'getInfoObject',
+                args: [Mock.Value.String, Mock.Value.Function],
```
Comment (dpobel) [inline Tests/js/models/assets/ez-locationmodel-tests.js]: same here

**Comment by dpobel** [comment]
there's another call to `loadRoot` in ez-locationviewviewservice.js that needs to replaced as well.

**Comment by yannickroger** [comment]
@dpobel I took care of your feedback.

**Comment by dpobel** [comment]
+1

**Pull request merged**